sql/2021/resource-hints/native_iframe_lazy_loading.sql

```diff
@@ -0,0 +1,41 @@
+#standardSQL
+# Count the number of lazily loaded iframes
+
+# returns true/false if the page has an iframe with `loading="lazy"` or null if the page has no iframes
+CREATE TEMPORARY FUNCTION hasLazyLoadedIframe(almanac_string STRING)
+RETURNS BOOL
+LANGUAGE js AS ''' 
+try {
+    var almanac = JSON.parse(almanac_string)
+    if (Array.isArray(almanac) || typeof almanac != 'object') return null;
+    
+    var iframes = almanac["iframes"]["iframes"]["nodes"]
+
+    if (iframes.length) {
+        return !!iframes.filter(i => (i.loading || "").toLowerCase() === "lazy").length
+    }
+
+    return null;
+}
+catch {
+    return null
+}
+''' ;
+
+SELECT
+  client,
+  COUNTIF(has_lazy_iframes) AS is_lazy,
+  SUM(COUNTIF(has_lazy_iframes IS NOT NULL)) OVER (PARTITION BY client) AS has_iframe,
+  COUNTIF(has_lazy_iframes) / SUM(COUNTIF(has_lazy_iframes IS NOT NULL)) OVER (PARTITION BY client) AS pct,
+  SUM(COUNT(0)) OVER (PARTITION BY client) AS total
```

Comment:
This query is grouped by client, so IIUC partitioning by client is a no-op.

```suggestion
  COUNTIF(has_lazy_iframes) AS is_lazy,
  COUNTIF(has_lazy_iframes IS NOT NULL) AS has_iframe,
  COUNTIF(has_lazy_iframes) / COUNTIF(has_lazy_iframes IS NOT NULL) AS pct,
  COUNT(0) AS total
```